convert2rhel/unit_tests/checks_test.py

```diff
@@ -474,6 +498,119 @@ def test_check_tainted_kmods(monkeypatch, command_return, expected_exception):
         check_tainted_kmods()
 
 
+@pytest.mark.parametrize(
+    # i.e. _bad_kernel_version...
+    ("any_of_the_subchecks_is_true",),
+    (
+        (True,),
+        (False,),
+    ),
+)
+def test_check_rhel_compatible_kernel_is_used(
+    any_of_the_subchecks_is_true,
+    monkeypatch,
+    caplog,
+):
+    monkeypatch.setattr(
+        checks,
+        "_bad_kernel_version",
+        value=mock.Mock(return_value=any_of_the_subchecks_is_true),
+    )
+    monkeypatch.setattr(
+        checks,
+        "_bad_kernel_substring",
+        value=mock.Mock(return_value=False),
+    )
+    monkeypatch.setattr(
+        checks,
+        "_bad_kernel_package_signature",
+        value=mock.Mock(return_value=False),
+    )
+    if any_of_the_subchecks_is_true:
+        with pytest.raises(SystemExit):
+            check_rhel_compatible_kernel_is_used()
+    else:
+        check_rhel_compatible_kernel_is_used()
+        assert "Kernel is compatible" in caplog.records[-1].message
+
+
+@pytest.mark.parametrize(
+    ("kernel_release", "major_ver", "exp_return"),
+    (
+        ("5.11.0-7614-generic", None, True),
+        ("3.10.0-1160.24.1.el7.x86_64", 7, False),
+        ("3.10.0-1160.24.1.el7.x86_64", 6, True),
+        ("5.4.17-2102.200.13.el8uek.x86_64", 8, True),
+        ("4.18.0-240.22.1.el8_3.x86_64", 8, False),
+    ),
+)
+def test_bad_kernel_version(
+    kernel_release, major_ver, exp_return, monkeypatch
+):
+    Version = namedtuple("Version", ("major", "minor"))
```

Comment:
Ideally all the data models should be moved to somewhere, from where we can import them. So we can do a better mocking, than this